### DISCUSSION TITLE
How to use OAuth2 with new version 3.0.2?

### DISCUSSION THREAD

**Discussion by jagdish-176** [oas-tools/oas-tools.github.io#11] [Q&A]
I have migrated from swagger-tools to oas-tools@2.x.x and now trying to migrate from 2.x to 3.x 

Previously with 2.x I have security handler as below where I could get token for verification
OAuth2 = async (req, def, scopes, next) =>{ /*validate token from req.Authorization */}

Now with 3.x, I'm unable to get token. I would appreciate quick help on this 
OAuth2: (secDef, secScope) => { /* OAuth authentication */ } // How do we get token here?

**Comment by alesancor1** [comment]
Are you sure you are using the right type of Authentication?

OAuth2 doesn't require a token in the `Authorization` header, it uses other methods to get the token, like an authorization URL (which is defined in the `secDef` object). Check the [OpenAPI documentation about OAuth2](https://swagger.io/docs/specification/authentication/oauth2).

If you want to use a token in the Authorization header consider using **HTTP basic** or **bearer** authentication types. Also, check out the [Authentication Middleware](https://github.com/oas-tools/oas-auth), which provides JWT implementation for bearer token.

**Reply by frogburger** [reply]
Actually you pass the access_ token to the API, this is the case for client credentials flows for example. So by just having the secdef and the scopes, makes the oauth2 security fairly useless... Systematically passing the req for the security handler will actually be extremely useful...
